rusoto/core/src/signature.rs

```diff
@@ -55,46 +66,57 @@ impl SignedRequest {
             canonical_uri: String::new(),
         }
     }
+    /// Adds a header of format <"content-type", content-type>
     pub fn set_content_type(&mut self, content_type: String) {
         self.add_header("content-type", &content_type);
     }
 
+    /// Sets the target hostname
     pub fn set_hostname(&mut self, hostname: Option<String>) {
         self.hostname = hostname;
     }
 
+    /// Sets the target hostname using the current service type and region
     pub fn set_endpoint_prefix(&mut self, endpoint_prefix: String) {
         self.hostname = Some(build_hostname(&endpoint_prefix, &self.region));
     }
 
+    /// Sets the new body (payload)
     pub fn set_payload(&mut self, payload: Option<Vec<u8>>) {
         self.payload = payload;
     }
 
+    /// Returns HTTP method
     pub fn method(&self) -> &str {
         &self.method
     }
 
+    /// Returns  path
```

Comment:
Besides adding "the current" to these , I also think we want to cut out this extra space :smile: